src/ImageSharp/GraphicOptionsDefaultsExtensions.cs

```diff
@@ -0,0 +1,100 @@
+        /// <param name="optionsBuilder">The action to update instance of the default options used.</param>
+        /// <returns>The passed in <paramref name="context"/> to allow chaining.</returns>
+        public static IImageProcessingContext SetGraphicsOptions(this IImageProcessingContext context, Action<GraphicsOptions> optionsBuilder)
+        {
+            var cloned = context.GetGraphicsOptions().DeepClone();
+            optionsBuilder(cloned);
+            context.Properties[typeof(GraphicsOptions)] = cloned;
+            return context;
+        }
+
+        /// <summary>
+        /// Sets the default options against the configuration.
+        /// </summary>
+        /// <param name="context">The image processing context to store default against.</param>
+        /// <param name="optionsBuilder">The default options to use.</param>
+        public static void SetGraphicsOptions(this Configuration context, Action<GraphicsOptions> optionsBuilder)
+        {
+            var cloned = context.GetGraphicsOptions().DeepClone();
+            optionsBuilder(cloned);
+            context.Properties[typeof(GraphicsOptions)] = cloned;
+        }
+
+        /// <summary>
+        /// Sets the default options against the image processing context.
+        /// </summary>
+        /// <param name="context">The image processing context to store default against.</param>
+        /// <param name="options">The default options to use.</param>
+        /// <returns>The passed in <paramref name="context"/> to allow chaining.</returns>
+        public static IImageProcessingContext SetGraphicsOptions(this IImageProcessingContext context, GraphicsOptions options)
+        {
+            context.Properties[typeof(GraphicsOptions)] = options;
+            return context;
+        }
+
+        /// <summary>
+        /// Sets the default options against the configuration.
+        /// </summary>
+        /// <param name="context">The image processing context to store default against.</param>
+        /// <param name="options">The default options to use.</param>
+        public static void SetGraphicsOptions(this Configuration context, GraphicsOptions options)
+        {
+            context.Properties[typeof(GraphicsOptions)] = options;
+        }
+
+        /// <summary>
+        /// Gets the default options against the image processing context.
+        /// </summary>
+        /// <param name="context">The image processing context to retrieve defaults from.</param>
+        /// <returns>The globaly configued default options.</returns>
+        public static GraphicsOptions GetGraphicsOptions(this IImageProcessingContext context)
+        {
+            if (context.Properties.TryGetValue(typeof(GraphicsOptions), out var options) && options is GraphicsOptions go)
+            {
+                return go;
+            }
+
+            var configOptions = context.Configuration.GetGraphicsOptions();
```

Comment:
Are we really sure we want the defaulting logic in the `Get***` extension methods, instead of populating `Properties` in the `DefaultImageProcessingContext` ctr?

What are the pros for this decision?

Reply:
Ah yeah... If our keys are strongly typed that makes comparison faster and easier too.

Reply:
Ah yeah... If our keys are strongly typed that makes comparison faster and easier too.

Reply:
FYI an interesting API proposal that also contains the concept of property bags: dotnet/runtime#1793

In total, a bit too verbose for my style, but I like the idea of having custom types for the bag, and (in some form) for the keys. I would also apply it for values, if it makes sense.

Reply:
I do kinda like the pattern proposed in the runtime but, (and it is a big but) it feels soo overengineered, sometimes (and in my experience usually) the simple solution is the right one.